Refactoring - second try

marekkalnik opened the pull request.

marekkalnik committed:
Re-refactor

marekkalnik committed:
Fix typo

marekkalnik committed:
Fixed stop/start following events bug

marekkalnik committed:
Disable forgotten stopLocate reference

domoritz commented:
Awesome, here are a few small things that need to be done before I merge:
- [ ] When following is enabled (as in the demo), and I do the following: click the (scrolls, icon is orange), pan the map so that the circle is outside the map view (icon is blue now) and then click the control again (to get the circle back into the view), the icon should now be orange and following enabled but it's not.
- [ ] Using the same procedure, I get a bunch of errors
  ![screen shot 2014-11-24 at 12 42 58](https://cloud.githubusercontent.com/assets/589034/5172777/8b6d4d56-73d7-11e4-96c3-3867ab4bb209.png)
- [ ] The fact that there are errors means that the circle looks funny if you move
  ![screen shot 2014-11-24 at 12 42 54](https://cloud.githubusercontent.com/assets/589034/5172788/adc335be-73d7-11e4-8b5c-5da2edd0e9a0.png)
- [ ] The readme needs a small section about how to extend the control

domoritz commented on src/L.Control.Locate.js:
```diff
@@ -130,238 +277,192 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
+            this._map.fire('startfollowing', this);
+            this._following = true;
+            if (this.options.stopFollowingOnDrag) {
+                this._map.on('dragstart', this._stopFollowing, this, this);
```
Why do you pass `this` two times?

gmccrackin replied:
I am also wondering about this.

marekkalnik replied:
It was introduced by error. Fixed.

gmccrackin commented on src/L.Control.Locate.js:
```diff
@@ -103,11 +103,158 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
+         * Used by drawMarker, you can ignore it if you have overriden it.
+         */
+        updateMarker: function(latlng, mStyle) {
+            this._marker.setLatLng();
```
@domoritz @marekkalnik The error looks like it is right here. The setLatLng function is not being passed the latlng variable.

marekkalnik replied:
Thanks @gmccrackin. Seems like passing the parameter has indeed fixed it.

gmccrackin replied:
np

marekkalnik commented:
Thanks for the feedback! I'll fix the errors this evening.

marekkalnik committed:
Fix missing parameter

marekkalnik committed:
Remove unnecessary parameter

marekkalnik committed:
Update doc to add info about extending

marekkalnik committed:
Remove uneeded parameters

marekkalnik committed:
Rename activate/deactivate and stop/start pairs

The public interface was actually made private so it should be the
opposite one

marekkalnik commented:
I have also added some doc, renamed the methods one would like to call to stop locating and removed some unnecessary parameters (especially in public methods).

I guess it should be ok now.

domoritz commented:
I was going over it. Will merge your latest changes and then commit to gh-pages. Thanks! This is great.

marekkalnik commented:
Great. Once it's done I'll try to port jtomaszewski's compass implementation to check if the API is exstensible enough.

domoritz commented:
Cool! I merged and changed a few minor things.
